sympy/algebras/quaternion.py

```diff
@@ -15,6 +16,18 @@
 from mpmath.libmp.libmpf import prec_to_dps
 
 
+def _check_norm(elements, norm):
+    """validate if input norm is consistent"""
+    if norm is not None and norm.is_number:
+        if not norm.is_positive:
```

Comment:
The more lenient `if norm.is_positive is False:` might be preferable in some cases where the sign is indeterminate. The test could be 

```python
def _check_norm(q, norm):
  if norm is not None;
    if norm.is_postive is False:  # norm could be vanilla x
        raise error
    numerical = all(i.is_number and i.is_real for i in elements)
    if numerical and is_eq(norm**2, sum(i**2 for i in elements)) is False:
        raise error
```

Reply:
Just did that!